property/index.js

```diff
@@ -115,54 +116,128 @@ module.exports = yeoman.generators.Base.extend({
-      name: this.name,
-      type: this.type
-    };
-    if (this.required) {
-      def.required = true;
-    }
-    if (this.defaultValue) {
-      if (this.type === 'boolean'){
-        if (['true', '1', 't'].indexOf(this.defaultValue) !== -1 ){
-          def.default = true;
-        } else {
-          def.default = false;
-        }
-      } else if (this.defaultValue === 'uuid' || this.defaultValue === 'guid'){
-         def.defaultFn = this.defaultValue;
-      } else if ((this.type === 'date' || this.type === 'datetime') &&
-                  this.defaultValue.toLowerCase() === 'now'){
-         def.defaultFn = 'now';
-      } else {
-         def.default = this.defaultValue;
-      }
-    }
-
-    this.modelDefinition.properties.create(def, function(err) {
-      helpers.reportValidationError(err, this.log);
+    this.modelDefinition.properties.create(this.propDefinition, function(err) {
+      helpers.reportValidationError(err, console.warn);
       return done(err);
     }.bind(this));
   },
-
   saveProject: actions.saveProject
 });
+
+function coerceDefaultValue(propDef, value) {
+  switch (propDef.type) {
+    case 'string':
+      if (value === 'uuid' || value === 'guid'){
+        propDef.defaultFn = value;
+      } else {
+        propDef.default = value;
+      }
+      break;
+    case 'number':
+      propDef.default = Number(value);
+      if(isNaN(propDef.default)) {
+        throw Error('Invalid default number value: '+ value);
+      }
+      break;
+    case 'boolean':
+      if (['true', '1', 't'].indexOf(value) !== -1 ){
+        propDef.default = true;
+      } else {
+        propDef.default = false;
+      }
+      break;
+    case 'date':
+      if (value.toLowerCase() === 'now'){
+        propDef.defaultFn = 'now';
+      } else {
+        var isNumber = /^[0-9]+$/.test(value);
+        if (isNumber) {
+          propDef.default = new Date(Number(value));
+        } else {
+          propDef.default = new Date(value);
+        }
+      }
+      break;
+    case 'array':
+      if (propDef.itemType === 'string') {
+        propDef.default = value.replace(/[\s,]+/g, ',').split(',');
+      } else if (propDef.itemType === 'number') {
+        propDef.default = value.replace(/[\s,]+/g, ',').split(',')
+          .map(function(item) {
+            return Number(item);
+          });
+      } else {
+        propDef.default = value;
+      }
+      break;
+    case 'geopoint':
+      if (value.indexOf('lat') !== -1 && value.indexOf('lng') !== -1) {
+        propDef.default = JSON.parse(value);
+      } else {
+        var geo = value.replace(/[\s,]+/g, ',').split(',');
+        propDef.default = {};
+        propDef.default.lat = Number(geo[0]);
+        propDef.default.lng = Number(geo[1]);
+      }
+      break;
+    case 'buffer':
+      console.warn('Warning: property default value was converted' + 
+        'from string using default node.js UTF8 encoding');
+      propDef.default = new Buffer(value);
```

Comment:
In our online discussion, we agreed to remove "default" prompt for buffer types.